📖♻️：diffuse agent forwarding content for openinf core in 2024 (#1106)
OpenINF/openinf.github.io: collections/_drafts/agent-forwarding.md

---
title: OpenINF Next-Gen Guidance on Agent Forwarding
category: contributing
authors: @OpenINF/wg-a-team

> publish: false

Author: ```suggestion
published: false
permalink:
layout:
```

🔗 https://jekyllrb.com/docs/front-matter/#predefined-global-variables

will get back to this in a few moments

---

> title: Agent Forwarding For OpenINF Core Members

Author: ```suggestion
title: Agent Forwarding for OpenINF Core in 2024
```

> authors: @OpenINF/ATeam

Author: ```suggestion
authors: @OpenINF/wg-a-team
```

publish: false
---

> ## 1     Agent forwarding

Author: ```suggestion
## Agent Forwarding

<br />
```

the numbers are nice-looking, but that would also result in the automatic creation of those IDs (not sure if intended)

store it in a gpg\-agent\-specific directory for later use. Once an SSH key has
been added to the gpg\-agent in this manner, the gpg\-agent will be ready to use
the newly-added key.

\[\[note\]\]

If the \`gpg\-agent\` receives a signature request, the user may need prompting
for a passphrase, which is necessary to decrypt any SSH keys stored. Since
the ssh\-agent protocol does not contain a mechanism for telling the agent on
which display/terminal it's running, gpg\-agent's ssh\-support will use the TTY
or X display where gpg\-agent started. To switch this display to the current
one, you may use the following command.

\`\`\`shell  
gpg\-connect-agent updatestartuptty /bye  
\`\`\`

\[\[/note\]\]

Although all GnuPG components try to start the \`gpg\-agent\` as needed, this is
not possible for the \`ssh\` support because \`ssh\` does not know about it.
Thus, if no GnuPG tool that usually accesses the \`gpg\-agent\` causing the
initial start of it ever ran, there is no guarantee that \`ssh\` can
use \`gpg\-agent\` for authentication. To fix this, you may
start \`gpg\-agent\` if needed using this simple command:

\`\`\`shell  
gpg\-connect-agent /bye  

> \`\`\`

Author: ~~~suggestion
```shell  
gpg-connect-agent /bye  
```
~~~

g'night for now btw, past my bedtime :p

---

> title: Agent Forwarding for OpenINF Core in 2024

Author: ~~~suggestion
---
title: OpenINF Next-Gen Guidance on Agent Forwarding
category: contributing
~~~

authors: @OpenINF/wg-a-team
publish: false

> ---

Author: ```suggestion
relevant_urls:
- https://github.com/OpenINF/docker-fisher/issues/5 # psusan + auth awkwardness
- https://dev.gnupg.org/T3883 # Win32-OpenSSH support for gpg-agent's ssh-agent
---
```

\[\[windows\]\]

However, there may be problems on Windows systems with Gpg4Win installed;
**Gpg4Win may have changed this default GnuPG home directory location** to an
AppData subdirectory (i.e., C:\\Users\\<username>\\AppData\\Roaming\\gnupg).
This location, however, is not the location our GPG agent will be using, so to
reaffirm our preference for the default location, we will set
the GNUPGHOME environment variable to ~/.gnupg in the Git Bash startup script by
running the following.

echo 'export GNUPGHOME="~/.gnupg"' >> .bashrc

\[\[endwindows\]\]

Setting environment variables

Add the following lines to your .bashrc or whatever initialization file is used
for all shell invocations:

GPG_TTY=$(tty)

export GPG_TTY

This variable may only be helpful if you use \`pinentry\-curses\` (the
terminal-based pin entry program).

\[\[windows\]\]

On Windows systems, you should add the above lines to the ~/.bashrc file for use
by Git Bash.

\[\[endwindows\]\]

GnuPG configuration

It's important to note that GnuPG on the remote system still needs your public
GPG keys to work correctly. So you have to ensure they are available on the
remote system even if your secret keys are not.

> \[\[note\]\]

Author: ```suggestion
\[\[note\]\]
```

btw, this cute lil admonition syntax was previously ruled as being non-preferable

> [&hellip;] should've been a directive.
>
> Square brackets are used a lot already in markdown [&hellip;]
> &mdash;https://twitter.com/wooorm/status/1701865524236345734

Author: <blockquote class="twitter-tweet"><p lang="en" dir="ltr">I personally have the same feedback as before: should&#39;ve been a directive.<br><br>Square brackets are used a lot already in markdown. The things they came up with conflict with a link definition ([!note]: &lt;&gt;)</p>&mdash; Titus 🇵🇸 (@wooorm) <a href="https://twitter.com/wooorm/status/1701865524236345734?ref_src=twsrc%5Etfw">September 13, 2023</a></blockquote>

Author: this would be a nice time for windings font selection

we'd have some diverse surrounding content and not all the text is uniform

🔗 https://dereknongeneric.github.io/agent-forwarding-eye-candy/agent-forwarding.html

Author: y'know, at first i was having trouble deciding
whether to keep the leading section numbers or
move that to some other place that doesn't interfere,
but now i really like how their addition makes it look academic

Author: btw, the HTML to Markdown conversion was done via online tool:

🔗 https://htmltomd.com/

briefly glancing at the output; it had stripped wayyy more tags than Pandoc
so, i was like "screw it, nobody else even comes close", but not sure why
all the backslashes have been applied to basically anything that can be

Author: re-reading https://github.com/OpenINF/GitHub-Markdown/issues/4 now and
gearing up for its demolition, which we are running
late, but kaboom happens in ~2hr 15min (10pm EST)

who like me still feels weird about using the `aside` HTML tag for them?

- http://html5doctor.com/understanding-aside/
- https://developers.google.com/style/notices
- https://docs.anaconda.com/restructuredtext/index.html#admonitions-notes

maybe i will feel better about it after reading those things on them

Author: last note worth mentioning on styleguide conventions
similar to those seen elsewhere: we used to
have our own freaking baby styleguide in our top-level
org dotrepo:

🔗 https://github.com/OpenINF/.github/commit/f720bce548a0a2a84c8a95f9f8a62528eb1b873f

in retrospect, i do not change my opinion about that being the wrong place for that handbook sub-project, which i was imagining would outgrow (in amount of content) the repo it had started in and the dot-prefix-namesake would soon have made no sense at all

regretting my decision of not migrating it else where tho

Author: ```suggestion
:::note{.note}
```

the actual resulting suggestion is better than this (gimme a sec to dig it up)

Author: > the actual resulting suggestion is better than this (gimme a sec to dig it up)

🔗 https://github.com/micromark/micromark-extension-directive?tab=readme-ov-file#syntax

`micromark-extension-directive` has definitely thought this out more clearly

---

ok, a few more to consider; would like to have this "override" title

- https://www.adamsdesk.com/posts/admonitions-jekyll/
- https://github.com/privacyguides/privacyguides.org/blob/1634bea9d2a66ef5804b68d18d4218e0b7a8b806/docs/meta/translations.md?plain=1#L2
- https://ericlikedis.github.io/posts/Admonitions-Documentation/
- https://github.com/lazee/premonition

the translations can happen in the document if necessary

> \[\[note\]\]
> 
> If you use VSCode with the remote extension pack, you may skip this step and
> move on to the next section. This step of copying over your local public GPG
> keyring into the remote container gets done automatically by the extension.
> 
> \[\[endnote\]\]

Author: ```suggestion
<aside class="note">

<b>ⓘ Note:</b>  
If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

</aside>
```

Author: this ~~cheap~~ inexpensive (frivolous) attempt above w/o using `picture` element (adaptive themes) was simply us trying on Google's not-very-good way of going about it

Author: > [&hellip;] should've been a directive.
>
> Square brackets are used a lot already in markdown [&hellip;]
> &mdash;https://twitter.com/wooorm/status/1701865524236345734

_Originally posted by @DerekNonGeneric in https://github.com/OpenINF/openinf.github.io/pull/1106#discussion_r1470135406_

🔗 https://talk.commonmark.org/t/generic-directives-plugins-syntax/444

that is wayyy too much reading, but skipping all of it would be fine, here is the one:

🔗 https://github.com/remarkjs/remark-directive?tab=readme-ov-file#example-styled-blocks

Author: ```suggestion
:::note{.note}

If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

:::
```

---
title: OpenINF Next-Gen Guidance on Agent Forwarding
category: contributing
authors: @OpenINF/wg-a-team
publish: false
---

> ## Agent Forwarding

Author: ```suggestion
```

there'd be no reason to have this here anymore since the document title would be included as part of the `deep-page` template

<br />

> 

Author: ```suggestion
```

there'd be no reason to have this here anymore since the document title would be included as part of the `deep-page` template

\[\[windows\]\]

However, there may be problems on Windows systems with Gpg4Win installed;
**Gpg4Win may have changed this default GnuPG home directory location** to an
AppData subdirectory (i.e., C:\\Users\\<username>\\AppData\\Roaming\\gnupg).
This location, however, is not the location our GPG agent will be using, so to
reaffirm our preference for the default location, we will set
the GNUPGHOME environment variable to ~/.gnupg in the Git Bash startup script by
running the following.

echo 'export GNUPGHOME="~/.gnupg"' >> .bashrc

\[\[endwindows\]\]

Setting environment variables

Add the following lines to your .bashrc or whatever initialization file is used
for all shell invocations:

GPG_TTY=$(tty)

export GPG_TTY

This variable may only be helpful if you use \`pinentry\-curses\` (the
terminal-based pin entry program).

\[\[windows\]\]

On Windows systems, you should add the above lines to the ~/.bashrc file for use
by Git Bash.

\[\[endwindows\]\]

GnuPG configuration

It's important to note that GnuPG on the remote system still needs your public
GPG keys to work correctly. So you have to ensure they are available on the
remote system even if your secret keys are not.

> \[\[note\]\]
> 
> If you use VSCode with the remote extension pack, you may skip this step and
> move on to the next section. This step of copying over your local public GPG
> keyring into the remote container gets done automatically by the extension.
> 
> \[\[endnote\]\]

Author: switching these up to use the styled block variation
of generic directives sytax

(this will all be very clearly explained in our own Markdown amalgamate's spec)

```suggestion
:::note{.note}

If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

:::
```

Refs: https://www.npmjs.com/package/remark-directive#example-styled-blocks

the HTML result is actually not exactly what we'd be outputting either:

```html
<div class="warning">
  <p>if you chose xxx, you should also use yyy somewhere…</p>
</div>
```
that should be wrapped in an `aside` tag? also missing the bold text and icon?

Author: <blockquote class="twitter-tweet"><p lang="en" dir="ltr">Markdown admonitions/badges, here we come. <a href="https://t.co/j07qwufKlH">https://t.co/j07qwufKlH</a></p>&mdash; FontVault (@FontVault) <a href="https://twitter.com/FontVault/status/1750744077337800742?ref_src=twsrc%5Etfw">January 26, 2024</a></blockquote>

Author: RE - ⬇️:

> re-reading https://github.com/OpenINF/GitHub-Markdown/issues/4 now and
> gearing up for its demolition, which we are running
> late, but kaboom happens in ~2hr 15min (10pm EST)
>
> who like me still feels weird about using the `aside` HTML tag for them?
>
> - http://html5doctor.com/understanding-aside/
> - https://developers.google.com/style/notices
> - https://docs.anaconda.com/restructuredtext/index.html#admonitions-notes
>
> maybe i will feel better about it after reading those things on them
>
> _Originally posted by @DerekNonGeneric in https://github.com/OpenINF/openinf.github.io/pull/1106#discussion_r1470413365_

GF - 🏛️:

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">opted for making old GF contiue her life in archive form<a href="https://t.co/J61RYALoyX">https://t.co/J61RYALoyX</a></p>&mdash; OpenINF (@OpenINF) <a href="https://twitter.com/OpenINF/status/1752173404289954250?ref_src=twsrc%5Etfw">January 30, 2024</a></blockquote>

Author: RE - ⬇️:

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">up next: we see about mirroring that yellow box as it should obv. be expressed in web tech for a custom admonitionn theme<a href="https://t.co/OB90Jk3bJL">https://t.co/OB90Jk3bJL</a></p>&mdash; OpenINF (@OpenINF) <a href="https://twitter.com/OpenINF/status/1752169541080084584?ref_src=twsrc%5Etfw">January 30, 2024</a></blockquote>

HW - 🏛️:

midnight eastern our classwork drop box self-destructs

i see y'all on codepen in ~30min!

🔗 https://codepen.io/DerekNonGeneric

Author: > RE - ⬇️:
> 
> > up next: we see about mirroring that yellow box as it should obv. be expressed in web tech for a custom admonitionn themehttps://t.co/OB90Jk3bJL— OpenINF (@OpenINF) [January 30, 2024](https://twitter.com/OpenINF/status/1752169541080084584?ref_src=twsrc%5Etfw)
> 
> HW - 🏛️:
> 
> midnight eastern our classwork drop box self-destructs
> 
> i see y'all on codepen in ~30min!
> 
> 🔗 https://codepen.io/DerekNonGeneric

📖⚠️：https://codepen.io/DerekNonGeneric/pen/GReQKvr

GPG Agent Configuration

There are a few configuration files needed for the operation of the agent. They
may all be found in the current GnuPG home directory, which defaults
to ~/.gnupg.

\[\[windows\]\]

However, there may be problems on Windows systems with Gpg4Win installed;
**Gpg4Win may have changed this default GnuPG home directory location** to an
AppData subdirectory (i.e., C:\\Users\\<username>\\AppData\\Roaming\\gnupg).
This location, however, is not the location our GPG agent will be using, so to
reaffirm our preference for the default location, we will set
the GNUPGHOME environment variable to ~/.gnupg in the Git Bash startup script by
running the following.

echo 'export GNUPGHOME="~/.gnupg"' >> .bashrc

\[\[endwindows\]\]

Setting environment variables

Add the following lines to your .bashrc or whatever initialization file is used
for all shell invocations:

GPG_TTY=$(tty)

export GPG_TTY

This variable may only be helpful if you use \`pinentry\-curses\` (the
terminal-based pin entry program).

\[\[windows\]\]

On Windows systems, you should add the above lines to the ~/.bashrc file for use
by Git Bash.

\[\[endwindows\]\]

> GnuPG configuration

Author: the ToC needs some thought for sure

```suggestion
### 1.4    GnuPG configuration
```

[[windows]]

On Windows systems, you should add the above lines to the ~/.bashrc file for use
by Git Bash.

[[endwindows]]

GnuPG configuration

It's important to note that GnuPG on the remote system still needs your public
GPG keys to work correctly. So you have to ensure they are available on the
remote system even if your secret keys are not.

:::note{.note}

If you use VSCode with the remote extension pack, you may skip this step and
move on to the next section. This step of copying over your local public GPG
keyring into the remote container gets done automatically by the extension.

:::

:::excerpt{.quote} <!-- TODO(DerekNonGeneric): link to source of below -->

<abbr title="Secure Copy Protocol">[SCP]{#scp .dfn}</abbr> is a means of
securely transferring [computer files][] between a local [host][] and a remote
host or between two remote hosts. It is based on the [Secure Shell][] (SSH)
protocol.[^1] "SCP" commonly refers to both the Secure Copy Protocol and the
program itself.[^2]

<!-- TODO(DerekNonGeneric): include how SCP's not:
- preferable over using SFTP for remote file transfer
- a bad choice for quick local transfers btwx host and guest vm/container(s)
-->

:::

> <!-- LINK LABEL DEFINITIONS: START -->
> 
> [computer files]: https://en.wikipedia.org/wiki/Computer_file
> [host]: https://en.wikipedia.org/wiki/Server_(computing)
> [Secure Shell]: https://en.wikipedia.org/wiki/Secure_Shell
> 
> [^1]: https://en.wikipedia.org/wiki/Secure_copy_protocol#cite_note-1
> [^2]: https://en.wikipedia.org/wiki/Secure_copy_protocol#cite_note-Pechanec-2)
> 
> <!-- LINK LABEL DEFINITIONS: END -->

Author: ```suggestion
<!-- LINK LABEL DEFINITIONS - START -->

[computer files]: https://en.wikipedia.org/wiki/Computer_file
[host]: https://en.wikipedia.org/wiki/Server_(computing)
[Secure Shell]: https://en.wikipedia.org/wiki/Secure_Shell

[^1]: https://en.wikipedia.org/wiki/Secure_copy_protocol#cite_note-1
[^2]: https://en.wikipedia.org/wiki/Secure_copy_protocol#cite_note-Pechanec-2)

<!-- LINK LABEL DEFINITIONS - END -->
```

☝️ after re-checking notes, have noticed a few inconsistencies here and elsewhere regarding the way blocked-off links are delimited w/ a comment&hellip;

it is a dash (one dash and Prettier docs are where we saw that initially); not a convention we have been good at following as most needs to be corrected, but going forward, let's not deviate from Prettier docs code examples lol (seriously, we might need to write a lint rule)

🔗 https://prettier.io/docs/en/ignore#range-ignore

:::

:::excerpt{.quote} <!-- TODO(DerekNonGeneric): link to article -->

<abbr title="Secure Copy Protocol">[SCP]{#scp .dfn}</abbr> is a means of
securely transferring [computer files][] between a local [host][] and a remote
host or between two remote hosts. It is based on the [Secure Shell][] (SSH)
protocol.[^1] "SCP" commonly refers to both the Secure Copy Protocol and the
program itself.[^2]

<!-- TODO(DerekNonGeneric): include how SCP's not:

> - intended for file transfer over hostile nets

Author: ```sugggestion
- preferred over SFTP for file transfer over hostile nets
```
